### PR TITLE
[IMP] im_livechat: introduce several PublicLivechatGlobal constants

### PR DESCRIPTION
**Pull request by alexkuhn**
Task-2899918

### CODE REVIEW

File: addons/im_livechat/static/src/legacy/public_livechat_history_tracking.js [hunk 1]
```diff
@@ -1,23 +1,8 @@
...
-if (!_.contains(urlHistory, page)) {
-    urlHistory.push(page);
-    while (urlHistory.length > HISTORY_LIMIT) {
-        urlHistory.shift();
-    }
-    utils.set_cookie(LIVECHAT_COOKIE_HISTORY, JSON.stringify(urlHistory), 60 * 60 * 24); // 1 day cookie
-}
+
 
 });
```
Comment (oomsveta) [inline addons/im_livechat/static/src/legacy/public_livechat_history_tracking.js:8]: is this file still useful?

File: addons/im_livechat/static/src/models/public_livechat_global.js [hunk 2]
```diff
@@ -1,8 +1,44 @@
...
+                }
+                utils.set_cookie(this.LIVECHAT_COOKIE_HISTORY, JSON.stringify(urlHistory), 60 * 60 * 24); // 1 day cookie
+            }
+        },
+    },
+    fields: {
+        LIVECHAT_COOKIE_HISTORY: attr({
+            default: 'im_livechat_history',
+        }),
+        HISTORY_LIMIT: attr({
```
Comment (oomsveta) [inline addons/im_livechat/static/src/models/public_livechat_global.js:33]: alphabetical

File: addons/im_livechat/static/src/models/public_livechat_global.js [hunk 3]
```diff
@@ -1,8 +1,44 @@
...
+            default: 15,
+        }),
+        RATING_TO_EMOJI: attr({
+            default: {
+                "5": "😊",
+                "3": "😐",
+                "1": "😞"
```
Comment (oomsveta) [inline addons/im_livechat/static/src/models/public_livechat_global.js:40]: missing trailing comma

Comment (oomsveta) [inline addons/im_livechat/static/src/models/public_livechat_global.js:40]: single quotes for keys?

File: addons/im_livechat/static/src/models/public_livechat_global.js [hunk 4]
```diff
@@ -1,8 +1,44 @@
...
+
+import utils from 'web.utils';
 
 registerModel({
     name: 'PublicLivechatGlobal',
     identifyingFields: ['messaging'],
+    lifecycleHooks: {
+        _created() {
+            // History tracking
+            const page = window.location.href.replace(/^.*\/\/[^/]+/, '');
```
Comment (oomsveta) [inline addons/im_livechat/static/src/models/public_livechat_global.js:14]: ➡️ `window.location.pathname`?

Comment (alexkuhn) [reply addons/im_livechat/static/src/models/public_livechat_global.js:14]: What do you mean by this change?
This LOC is just a move, so I'm not entirely sure it's equivalent to `window.location.pathname`.

The "?" at end of your comment gives me the impression you are also unsure.

Keep in mind lots of LOCs in public livechat I still don't get them.
In the meantime, I do structural changes (i.e. basically moves and removal of dead/useless code), so if we are unsure, better keep the exact same LOCs.

**Comment by alexkuhn** [comment]
All changes done. Thanks!

**Review by oomsveta — APPROVED** [review]

**Comment by alexkuhn** [comment]
Merged in grouped PR: https://github.com/odoo/odoo/pull/96690

**Pull request closed without merging**